### DISCUSSION TITLE
L-BFGS updated P matrix

### DISCUSSION THREAD

**Discussion by LiuZhexuan** [PREDICT-EPFL/piqp#4] [Q&A] [answered]
I want to find a subproblem solver for my SQP solver. Target problems are too complicated to provide matrix P since it requires second order derivatives for the original problem. Is there a way to use L-BFGS updated P matrix?
Otherwise I have to give dense P matrix to QP subproblem solver and it may be impossible to solve.

**Comment by RSchwan** [comment, marked as answer]
As far as I understand, L-BFGS doesn't construct the Hessian matrix explicitly. Hence, I assume you are considering the normal BFGS algorithm. One problem with BFGS is that in the standard algorithm, the updates result in a dense P matrix. Hence, you probably need some kind of sparsity preserving L-BFGS update. As long as the sparsity pattern doesn't change (which it doesn't in the dense case), you can update the P matrix without internal memory allocations. What kind of problem are you trying to solve?

**Reply by LiuZhexuan** [reply]
Yes, if the QP solver cannot deal with L-BFGS, I have to use BFGS and store the whole P matrix. I'm not quite familiar with the sparsity preserving BFGS update but I will do some search, thanks!
My problem is about chemical process optimization and usually have less than 10^5 variables with <10^3 degrees of freedom. Now I'm considering whether to use a open source QP solver or to implement a null space method (recommended by <Numerical Optimization 2nd> for number of degrees of freedom is small)
